### PR TITLE
Don't attempt to parse HTML as JS in sourcemap processor

### LINKED ISSUE #3672: Don't attempt to parse HTML documents as JS
We have observed more than a few times that some websites are returning 200 OK for "Not Found" HTML documents. As a result, Sentry thinks this is a JavaScript file and attempts to parse it / perform sourcemap transforms. This leads to issues like this:

image

This isn't a straight forward solution, because what if you have an error originating from a 404 page? That page could have inline `<script>` tags where the error is occurring, so in that case we'll want to parse the page and apply it. We could maybe discriminate a little more depending on the type of file (i.e. JS vs HTML), but even then, we can't guarantee the "type" of a file based on extension or content-type.

### PR DESCRIPTION
**Pull request by benvinegar**
Fixes #3672 

I've had 2 people in the last week run into this. Rather than give them a bunch of "Invalid Location" found errors, I think it will be helpful (and less busy, UI-wise) to tell them the content of the file was wrong.

cc @getsentry/infrastructure

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/3754)

### CODE REVIEW

File: src/sentry/lang/javascript/processor.py [hunk 1]
```diff
@@ -366,6 +366,21 @@ def fetch_file(url, project=None, release=None, allow_scraping=True):
         }
         raise CannotFetchSource(error)
 
+    # For JavaScript files, check if content is something other than JavaScript/JSON (i.e. HTML)
+    # NOTE: possible to have JS files that don't actually end w/ ".js", but this should catch 99% of cases
+    if url.endswith('.js'):
+        # Check if response is HTML by looking if the first non-whitespace character is an open tag ('<').
+        # This cannot parse as valid JS/JSON.
+        # NOTE: not relying on Content-Type header because apps often don't set this correctly
+        body_start = result[1][:20].strip()  # Discard leading whitespace (often found before doctype)
```
Comment (mattrobenolt) [inline src/sentry/lang/javascript/processor.py]: `lstrip()`

File: src/sentry/lang/javascript/processor.py [hunk 2]
```diff
@@ -366,6 +366,21 @@ def fetch_file(url, project=None, release=None, allow_scraping=True):
...
+
+        if body_start[:1] == '<':
+            error = {
+                'type': EventError.JS_INVALID_CONTENT,
+                'url': url
```
Comment (mattrobenolt) [inline src/sentry/lang/javascript/processor.py]: Trailing comma plz.

**Comment by mattrobenolt** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3754)*

**Comment by dcramer** [comment]
Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3754)*

**Pull request merged**